src/main/core/manager.rs

```diff
@@ -516,6 +524,17 @@ impl<'a> Manager<'a> {
             });
         }
 
+        let stats_filename = self.data_path.clone().join("sim-stats.json");
+        let stats_file = std::fs::File::create(&stats_filename)
+            .with_context(|| format!("Failed to create file '{}'", stats_filename.display()))?;
+
+        serde_json::to_writer_pretty(stats_file, &sim_stats).with_context(|| {
+            format!(
+                "Failed to write stats json to file '{}'",
+                stats_filename.display()
+            )
+        })?;
+
```

Comment:
I would prefer to push this code inside of `sim_stats.rs` to try to reduce complexity as much as possible in the already-complex manager. Is there a good argument against doing that?

Reply:
I moved this to `sim_stats`.